### ISSUE TITLE
Light version for non-security use cases

### ISSUE THREAD

**Issue by skiinggood** [Valve/fingerprintjs2#172]
Propose that the project could have an minimal size alternative build optimized around device identification for when consumers have NOT taken steps to block conventional persistent state management (ie disabled cookies/LSO's, expressed DNT, or installed blockers).   

Rationale for a "light" use case:
1) When useragents support persistent state management (eg cookies), then fingerprints are LESS useful; 
and 2) a user's explicit choice to prevent third party persistent state management (eg blocking cookies) SHOULD be respected; 
Therefore: 3) useragents that by DEFAULT block cookies (subjective: "and are not typically installed by user primarily for blocking") drive the need for statistical device ids.  

Conceptually a "light" version would be optimized for case 3) which would today mean primarily SAFARI and OPERA and might have the following characteristics:
a) Exclude code for the tampering and Flash options 
b) Pruned set of fonts and plugins to detect (i.e., no IE).  
c) Exclude the Canvas option because of its bulk and performance impact
d) Include simple useragent parsing given the constrained format of Safari and Opera.

Thinking about tackling this on my own but wondered if this would be useful as a part of the v2.0 effort which already includes removing Flash detection?  Also want to make sure this isn't redundant with another effort.

-- "skiing...good"

**Comment by Valve** [comment]
Good idea. Why are you not considering just disabling certain "heavy" options?
Canvas not being one of them, btw. The heaviest is font detection with JS.

**Comment by skiinggood** [comment]
a) and b) drive it.   
a) Because I think JS font and plugin detection "may" be important on Safari/Opera and so would need to "prune" those options or create alternate versions to exclude significant IE logic and fonts that never occur on Safari/Opera?
b) Generic UserAgent parsing is fairly large and performance intensive.  I think for Safari and Opera heuristics can be used (e.g., lop off last 5 bytes) to create adequate stability.

**Comment by Valve** [comment]
JS font detection is already split into "basic" and "extended".
As for UA parsing - this topic has popped up multiple times in this project and it was generally decided that it will be delegated to external code (e.g. a plugin). So in v2 there will be no Flash integration, possibly updated, slimmer "basic" subset of fonts for detection and ability to pre-process UA string before hashing it.

**Comment by skiinggood** [comment]
Perhaps the JS font can be further trimmed by a useragent/OS specific filter?
Does the "basic" version exclude the activeX/IE interrogation?

The primary rationale is that a Safari/Opera constrained version can be more aggressively pruned and that hard problems like useragent parsing can simplified.  The big driver is download size.  The full FingerprintJS2 is 60k vs 10k for FingerprintJS which without Canvas minifies down to less than 3k.

Re: Canvas: 
Canvas has some code and performance impact and more importantly has had high profile negative attention.

**Comment by Valve** [comment]
Ok, thanks for letting me know will think about it
